### PR TITLE
doc news: add 14.0.3 entry

### PR DESCRIPTION
**Pull request by komainu8**

### CODE REVIEW

**Comment by otegami** [comment]
```diff
-  `request_cancel`でクエリーを中断するとクラッシュすることがある問題を修正しました。
-
 * Supress grn_output_range_normalize failed

-  以下のように、`--post_filter`実行時に`--offset`の値が範囲外になった場合にエラーになる問題を修正しました。
-  `--filter`と`--offset`の組み合わせでは、エラーにならないため、`--filter`と`--post_filter`で挙動が異なっていました。
+  Fixed the unexpected error when using `--post_filter` with `--offset` greater than the post-filtered result.
+  In the same situation, using `--filter` with `--offset` doesn't raise the error.
+  This inconsistency in behavior between `--filter` and `--post-filter` has now been resolved.

-  (具体例)
+  table_create Users TABLE_PAT_KEY ShortText
+  column_create Users age COLUMN_SCALAR UInt32
+  load --table Users
+  [
+    ["_key", "age"],
+    ["Alice", 21],
+    ["Bob", 22],
+    ["Chris", 23],
+    ["Diana", 24],
+    ["Emily", 25]
+  ]
+  select Users \
+    --filter 'age >= 22' \
+    --post_filter 'age <= 24' \
+    --offset 3 \
+    --sort_keys -age
+  [[0,0.0,0.0],[[[3],[["_id","UInt32"],["_key","ShortText"],["age","UInt32"]]]]]

-* 近傍フレーズ直積検索で`(...)`内のすべてのフレーズがマッチしない場合に、誤った検索結果を返す場合がある問題を修正しました。
+* Fixed a bug where incorrect search result could be returned when not all phrases within `(...)` matched using Near phrase product.

-  例えば以下の、`--query '*NPP1"(a) (2)"'`で指定している`(2)`にマッチするレコードがありません。
-  この場合は、何もヒットしないのが正しい挙動ですが、`--query '*NPP1"(a)`相当の挙動になっていました。
-  つまり、`(2)`にマッチするレコードが無いにも関わらず、`ax1`と`axx1`がヒットしていました。
+  For example, there is no record which matched `(2)` condition using `--query '*NPP1"(a) (2)"'`.
+  In this case, the expected behavior would be return no record. However, the actual behavior was equal to the query `--query '*NPP and "(a)"`.
+  This means that despite no records matched `(2)`, records like `ax1` and `axx1` were incorrectly returned.
```

File: doc/source/news/14.md [hunk 1]
```diff
@@ -1,5 +1,351 @@
...
+  ]
+  select Users \
+    --filter 'age >= 22' \
+    --post_filter 'age <= 24' \
+    --offset 3 \
+    --sort_keys -age
+  [[0,0.0,0.0],[[[3],[["_id","UInt32"],["_key","ShortText"],["age","UInt32"]]]]]  
+  ```
+
+* Fixed a bug where incorrect search result could be returned when not all phrases within `(...)` matched using Near phrase product.
```
Comment (otegami) [inline doc/source/news/14.md:129]: I'm so sorry I did typo here.
```suggestion
* Fixed a bug where incorrect search result could be returned when not all phrases within `(...)` matched using near phrase product.
```

File: doc/source/news/14.md [hunk 2]
```diff
@@ -1,5 +1,388 @@
...
+      }
+    ]
+  ]
+  ```
+
+* [{doc}`/install/ubuntu`] Add support for Ubuntu 24.04 LTS (Noble Numbat).
+
+### Fixes
+
+* [{doc}`/reference/commands/request_cancel`] Fix a bug that Groonga may crash when we execute `request_cancel` command whlie we execute the other query.
```
Comment (abetomo) [inline doc/source/news/14.md:101]: > whlie

while?

File: doc/source/news/14.md [hunk 3]
```diff
@@ -1,5 +1,388 @@
...
+      ]
+    ]
+  ]
+  ```
+
+* Fixed a bug that rehash failed or data in a table broke when rehash occured that the table with `TABLE_HASH_KEY` has  2^28 or more records.
+
+* Fixed a bug that highlight position slipped out of place in the following cases.
+
+  * If full width space existed before highlight target charcters as below.
```
Comment (abetomo) [inline doc/source/news/14.md:196]: > charcters

characters?

**Comment by komainu8** [comment]
@abetomo Thank you for your comments. I've fixed them.

**Pull request merged**